portal-kernel/src/com/liferay/portal/kernel/comment/DiscussionPermission.java

```diff
@@ -53,6 +53,10 @@ public default boolean hasPermission(Comment comment, String actionId)
 	public boolean hasPermission(long commentId, String actionId)
 		throws PortalException;
 
+	public boolean hasPermission(
+		String actionId, String className, long classPK, long companyId,
+		long groupId);
+
 	public boolean hasSubscribePermission(
```

Comment:
@W2jx, if we must add the new method, it seems we should make the param **String actionId** as the last param.

Reply:
Hi @yuhai , Don't we need to sort by the first letter of the field name?